### DISCUSSION TITLE
加入 Pocket 后自动保存到简悦的稍后读（包含快照）

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3024] [服务]
> 如果你是 Pocket 用户，可以利用 Zapier 和 简悦 API，将有需要的内容导入到简悦，优势：

1. 此方式会在你的坚果云（或 Dropbox）生成本地快照。
2. 接入到简悦后，就可以使用简悦的各种「导出大礼包」与更专业的标注服务。

# 同属与「稍后读」为什么简悦并不排斥同类产品

> 如果你是 Hypothesis 的用户，自然知道简悦也可以 [导入到 Hypothesis](https://github.com/Kenshin/simpread/discussions/2085#discussioncomment-926749)。

> 其根本原因是：简悦是真正意义的不绑架用户数据，所以**简悦并不会像其它应用一样，限制用户使用简悦的方式**。

# 前置知识

> 需要事先完成下面的配置。

1. [如何开通开放平台](https://github.com/Kenshin/simpread/discussions/2757)
2. [API 快照功能](https://github.com/Kenshin/simpread/discussions/2994)
3. [如何让稍后读支持读取本地缓存](https://github.com/Kenshin/simpread/discussions/2146)

# Zapier

> 此方式需要使用 Zapier 这个自动化工具，如果你不了解它的话，可以类比是 IFTTT，但比 IFTTT 提供更强大 / 灵活的自动化定制。

# 如何配置

> 如果你使用过 Zapier，整个过程不会超过五分钟，否则请根据下图的一步步操作即可。

> 请务必使用此 [邀请链接](https://zapier.com/developer/public-invite/121575/cbaa1ac9bb7fa64f024515b7f46d5e1b/) 进入才能看到简悦的服务。

![image](https://user-images.githubusercontent.com/81074/140700217-2781436a-9890-4988-885e-06fb8bb0a07b.png)

![image](https://user-images.githubusercontent.com/81074/140700228-19a392ba-eb38-4b7a-b646-d8db779f3959.png)

![image](https://user-images.githubusercontent.com/81074/140700236-341062be-3263-429f-9a15-ff47cf1c0cb4.png)

> 注意，在选择 SimpRead 时，有两个版本：

 - 1.0.0 → 此版本中 `title` 是必选项。（ 此版本没有快照功能）
 - 1.0.1 → 此版本中 `title` 可不选（不选择 `title` 时，会自动保存快照）
 
> 下面的截图都是 **1.0.0** （可更改为 **1.0.1**）

![image](https://user-images.githubusercontent.com/81074/140700244-7717ddf7-fc21-4d1d-8cf5-ebad43814719.png)

![image](https://user-images.githubusercontent.com/81074/140700251-fe1ce10a-0384-4a85-8690-b6df89ee0e50.png)

![image](https://user-images.githubusercontent.com/81074/140700261-f7131cab-b306-40f1-85fc-e117c5baf5ac.png)

![image](https://user-images.githubusercontent.com/81074/140700265-fb47d19f-7803-40ad-b719-8b66f2c581d1.png)

![image](https://user-images.githubusercontent.com/81074/150916068-775b3061-9f40-4d09-8651-a92b00d639c8.png)

![image](https://user-images.githubusercontent.com/81074/140700276-68db6da5-3e3f-47f0-808e-38174822dd7e.png)

![image](https://user-images.githubusercontent.com/81074/140700283-c87ea6df-2e02-4687-aa04-63170519afb3.png)

![image](https://user-images.githubusercontent.com/81074/140700290-208d354f-c348-43ce-820e-66847bf0d57a.png)

# 不选择 `title` 时会保存快照

![image](https://user-images.githubusercontent.com/81074/140703455-c882425b-c744-408d-8d1e-3f9b61ae6053.png)

# 当加入特定标签后，才会保存到简悦中

> 可以选择 **New Tagged Item**

![image](https://user-images.githubusercontent.com/81074/140701051-21fc2265-99d0-4317-9b64-fb5df4099b69.png)

![image](https://user-images.githubusercontent.com/81074/140701181-e04fcef0-83de-44a2-ad8c-49c57b745575.png)

![image](https://user-images.githubusercontent.com/81074/140702401-fe2757cb-53d2-4815-ac2e-ceddee055c16.png)

# 关联

- [利用 Airtable 实现属于自己的链接公开分享集 #3003](https://github.com/Kenshin/simpread/discussions/3003)
- [Raindrop.io 加星后自动导入到简悦并生成本地快照 #3021](https://github.com/Kenshin/simpread/discussions/3021)
- [Instapaper 加星后自动保存到简悦的稍后读（包含快照） #3034](https://github.com/Kenshin/simpread/discussions/3034)
- [将简悦变成你的 RSS 阅读器（自动下载快照到本地） #3036](https://github.com/Kenshin/simpread/discussions/3036)
- [保存到 Pinboard.in 同时也保存到简悦（包括本地快照） #3048](https://github.com/Kenshin/simpread/discussions/3048)
- [Inoreader 加星后自动保存到简悦的稍后读以及生成本地快照 #3007](https://github.com/Kenshin/simpread/discussions/3007)
- [订阅 Newsletter 后自动保存到简悦的稍后读 #3088](https://github.com/Kenshin/simpread/discussions/3088)
- [当加入稍后读后，自动发送到你的 Telegram Channel #3092](https://github.com/Kenshin/simpread/discussions/3092)
- [发送当前 URL 到飞书群（包括：加入稍后读或手动触发）#3104](https://github.com/Kenshin/simpread/discussions/3104)
- [保存 RSS 的全文到简悦的稍后读（包括快照） #3186](https://github.com/Kenshin/simpread/discussions/3186)

# 引申

- [[教程\]如何开通开放平台 #2757](https://github.com/Kenshin/simpread/discussions/2757)
- [当在开放平台设置完毕后，如何测试配置是否成功？ #3020](https://github.com/Kenshin/simpread/discussions/3020)
- [授权登录（如何使用第三方登录简悦的高级账户） #1570](https://github.com/Kenshin/simpread/discussions/1570)
- [个人中心 #2959](https://github.com/Kenshin/simpread/discussions/2959)
- [免密码登录 #2974](https://github.com/Kenshin/simpread/discussions/2974)

**Comment by scramjetFoam** [comment]
您好，
现在部署时，setup action选项里没有1.Top Image URL：。
不知道怎么回事？

**Reply by Kenshin** [reply]
其它有吗？这个不是必选的项目。

这个选项是根据 URL 而变化的，或许你的测试 URL 数据中没有这个选项。

**Reply by Kenshin** [reply]
嗨，没问题就好，有问题的话，请通过  https://github.com/Kenshin/simpread/discussions/3024 直接过来提问。

PS：不要通过 Email 回复，它会再新建一个  😂

**Comment by shiningliang** [comment]
您好，在2. Create New Unreader in SimpRead (1.0.0)时
Set up action，Insert data下只有一个1. New Item in Pocket 选项
图示这种是手打的还是我配置Pocket不对呢

**Reply by Kenshin** [reply]
应该没问题的，最好截图看下。

**Reply by shiningliang** [reply]
<img width="1680" alt="截屏2022-01-24 23 56 26" src="https://user-images.githubusercontent.com/11460366/150818503-4a798f7d-22f0-4848-b840-f3dc09324349.png">

**Reply by Kenshin** [reply]
估计你的 Pocket 里面没有任何记录吧？

**Reply by shiningliang** [reply]
![image](https://user-images.githubusercontent.com/11460366/150898625-8781b251-4842-4182-acb8-9c6933f9e475.png)

有东西来着，我还特地加了几个做测试。。。我注销一下zapier试试

**Reply by shiningliang** [reply]
Okay了，原来Pocket那最后一步有一个Test Trigger要点一下，才能解析出一个case，例图里没有这一步

**Reply by Kenshin** [reply]
> Okay了，原来Pocket那最后一步有一个Test Trigger要点一下，才能解析出一个case，例图里没有这一步

好，我一会补上这个图。

**Reply by shiningliang** [reply]
大佬辛苦了

**Reply by Kenshin** [reply]
小事情。

另外，建议如果需要快照的话，请使用 1.0.1 （教程中有提到）

**Reply by Kenshin** [reply]
图已经补上了。

**Comment by tinainsz** [comment]
你好，用了一段时间这个工作流，非常顺手，感谢。不过昨天开始zapier突然出现警告，也同步不了，请问是api出现什么问题了吗？
![D54FE309-8B93-4F59-8827-993561F01EAB](https://user-images.githubusercontent.com/44186001/154195176-2fb9752c-7037-47eb-b853-e77e21d32de4.jpeg)

**Reply by Kenshin** [reply]
嗨，我刚才测试了下，没有问题的。

你可以重新添加一个自动化，如 **加星的 Pocket 导入到简悦**，如下图（只是用来测试你那边的简悦是否正确而已）

![image](https://user-images.githubusercontent.com/81074/154223832-9df0f6e1-2021-4d4a-8192-46120a475fb9.png)

然后测试到下图效果，确保返回 201，你的稍后读能获取到这个快照，就说明没问题。

![image](https://user-images.githubusercontent.com/81074/154223984-119ab8f4-bace-4775-8926-7cbee7d97a62.png)

假设这个测试没问题的话，如果原来的仍有问题，可以试试删除原来的 zap 重新再添加一次。

另外注意：看看是不是你的坚果云容量超了造成的？

**Reply by Kenshin** [reply]
via @tinainsz 

你好，感谢回复！
我刚刚又看了一下，很奇怪，它又自己好了……😂️
![image](https://user-images.githubusercontent.com/44186001/154264002-44d50e6c-7668-4adb-9708-abe11ad2c974.png)

**Reply by Kenshin** [reply]
嗨，没问题就好。

zapier 毕竟是免费版，偶尔抽风正常。

**Comment by Kenshin** [comment]
您好，我创建的动作测试显示成功了，但是稍后读里并没有新增未读
![image](https://user-images.githubusercontent.com/25945037/207910786-75fde24f-a11f-4ffd-b2c9-242d1d837ff6.png)

_Originally posted by @wanghxhx in https://github.com/Kenshin/simpread/discussions/3024#discussioncomment-4411509_

**Reply by Kenshin** [reply]
@wanghxhx 

嗨，我把你的原来的提问误删了，又帮你重新建立了一个回复。😂

***

1. 这个过程并不是马上生效，有个时间差，一般在 15分钟左右。
2. 你可以看看坚果云 Web 端里面的 `simpread_config.json` 是否更新了？
3. 如果更新了，但你的稍后读里面没有的话，说明你可能没有开坚果云客户端，你可以直接看 [这个教程](https://www.yuque.com/kenshin/simpread/wkswh7) 来配置。

**Reply by wanghxhx** [reply]
好了好了，是我没有设置自动同步，手动同步一下就收到了，谢谢啦！！

**Reply by Kenshin** [reply]
> 好了好了，是我没有设置自动同步，手动同步一下就收到了，谢谢啦！！

Cool，手动同步也一样~（相当于按需同步。

**Comment by ylzty** [comment]
很迷惑，zap history里显示任务成功了，但是简悦的稍后读里并没有显示。

![111](https://user-images.githubusercontent.com/78536614/233651905-d3111fb1-1d71-463f-8578-52501aff81a6.png)
![222](https://user-images.githubusercontent.com/78536614/233651920-76fd3f52-bb0d-4b57-8a28-029a345e2f84.png)

**Reply by Kenshin** [reply]
1. 它的 API 比较慢，可能需要多等一段时间。
2. 确认下坚果云 Web 端与你的本地 `simpread_config.json` 是否更新了？
3. 稍后读中读取本地快照的功能是否开了？
